tensorly/contrib/decomposition/_tt_cross.py

```diff
@@ -94,9 +94,7 @@ def tensor_train_cross(input_tensor, rank, tol=1e-4, n_iter_max=100, random_stat
     if isinstance(rank, int):
         rank = [rank] * (tensor_order + 1)
     elif tensor_order + 1 != len(rank):
-        message = "Provided incorrect number of ranks. Should verify len(rank) == tl.ndim(tensor)+1, but len(rank) = {} while tl.ndim(tensor) + 1  = {}".format(
-            len(rank), tensor_order
-        )
+        message = f"Provided incorrect number of ranks. Should verify len(rank) == tl.ndim(tensor)+1, but {len(rank) = } while tl.ndim(tensor) + 1  = {tensor_order}"
```

Comment:
Changed to support Python 3.7 and shortened line

```suggestion
        message = (
            "Provided incorrect number of ranks. Should verify "
            + "len(rank) == tl.ndim(tensor)+1, but len(rank) = {len(rank)} "
            + "while tl.ndim(tensor) + 1  = {tensor_order}"
        )
```

Reply:
You forgot the `f` for the f-strings

Reply:
```suggestion
        message = (
            "Provided incorrect number of ranks. Should verify "
            + f"len(rank) == tl.ndim(tensor)+1, but len(rank) = {len(rank)} "
            + f"while tl.ndim(tensor) + 1  = {tensor_order}"
        )
```